module/network/RobotCommunication/src/RobotCommunication.cpp

```diff
@@ -0,0 +1,75 @@
+#include "RobotCommunication.hpp"
+
+#include "extension/Configuration.hpp"
+
+#include "message/input/GameState"
+#include "message/input/RoboCup.hpp"
+#include "message/localisation/FilteredBall.hpp"
+// add includes
+
+namespace module::network {
+
+    using extension::Configuration;
+    using message::input::RoboCup;
+    // add usings
+
+    RobotCommunication::RobotCommunication(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)) {
+
+        on<Configuration>("RobotCommunication.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file RobotCommunication.yaml
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+            // need to determine send and receive ports
+            send_port = config["send_port"].as<uint>();
+            // need to determine broadcast ip
+            broadcast_ip = config["broadcast_ip"].as<uint>();
+
+            // If we are changing ports (the port starts at 0 so this should start it the first time)
+            if (config["receive_port"].as<uint>() != receive_port) {
+                // If we have an old binding, then unbind it
+                // The port starts at 0 so this should work
+                if (recieve_port != 0) {
+                    listenHandle.unbind();
+                }
+
+                receive_port = config["receive_port"].as<uint>();
+
+                // Bind our new handle
+                std::tie(listen_handle, std::ignore, std::ignore) =
+                    on<UDP::Broadcast, Single>(port).then([this](const UDP::Packet& p) {
+                        emit(NUClear::util::serialise::Serialise<SensorMeasurements>::deserialise(p));
```

Comment:
```suggestion
                        emit(NUClear::util::serialise::Serialise<RoboCup>::deserialise(p));
```
oops copy paste error on my part. this is where we get the data from other robots and emit it to the system for use. might want to put a log here so you can check we are receiving properly.